source/common/http/hash_policy.cc

```diff
@@ -39,14 +42,39 @@ class HeaderHashMethod : public HashMethodImplBase {
                                     const StreamInfo::FilterStateSharedPtr) const override {
     absl::optional<uint64_t> hash;
 
-    // TODO(mattklein123): Potentially hash on all headers.
     const auto header = headers.get(header_name_);
     if (!header.empty()) {
+      const bool check_multiple_values =
+          Runtime::runtimeFeatureEnabled("envoy.reloadable_features.hash_multiple_header_values");
```

Comment:
@mattklein123 @jmarantz 

After switch back to xxhash, it can get the same hash value just like the original XXH64 method. I tested that

`
TEST(Hash, xxHash) {
  EXPECT_EQ(3728699739546630719U, HashUtil::xxHash64("foo"));
  EXPECT_EQ(5234164152756840025U, HashUtil::xxHash64("bar"));
  EXPECT_EQ(8917841378505826757U, HashUtil::xxHash64("foo\nbar"));
  EXPECT_EQ(4400747396090729504U, HashUtil::xxHash64("lyft"));
  EXPECT_EQ(17241709254077376921U, HashUtil::xxHash64(""));
}

TEST(Hash, xxHashWithVector) {
  {
    absl::InlinedVector<absl::string_view, 1> v{"foo"};
    EXPECT_EQ(3728699739546630719U, HashUtil::xxHash64(v));
  }
  {
    absl::InlinedVector<absl::string_view, 1> v{"bar"};
    EXPECT_EQ(5234164152756840025U, HashUtil::xxHash64(v));
  }
  {
    absl::InlinedVector<absl::string_view, 1> v{"foo\nbar"};
    EXPECT_EQ(8917841378505826757U, HashUtil::xxHash64(v));
  }
  {
    absl::InlinedVector<absl::string_view, 1> v{"lyft"};
    EXPECT_EQ(4400747396090729504U, HashUtil::xxHash64(v));
  }
  {
    absl::InlinedVector<absl::string_view, 1> v{""};
    EXPECT_EQ(17241709254077376921U, HashUtil::xxHash64(v));
  }

  absl::InlinedVector<absl::string_view, 1> v{"foo", "bar"};
  EXPECT_EQ(17745830980996999794U, HashUtil::xxHash64(v));
  absl::InlinedVector<absl::string_view, 1> empty_vector;
  EXPECT_EQ(17241709254077376921U, HashUtil::xxHash64(empty_vector));
}`

So the question is, do we still need this runtime flag? since the user won't feel any different or any behavior change.

Reply:
Yes we still need the flag because we may now hash multiple values vs. 1.

Reply:
ah, right, the user's client may already sent multiple values, so we still need this flag.